app/views/service_worker/index.js.erb

```diff
@@ -37,40 +37,45 @@
   function createPageStream(request) {
     const stream = new ReadableStream({
       start(controller) {
-        if (!caches.match('/shell_top') || !caches.match('/shell_bottom')) { //return if shell isn't cached.
-          return
-        }
+        Promise.all([caches.match('/shell_top'), caches.match('/shell_bottom')])
+          .then((cachedShellMatches) => {
+            const cachedShellTop = cachedShellMatches[0];
+            const cachedShellBottom = cachedShellMatches[1];
+            if (!cachedShellTop || !cachedShellBottom) { // return if shell isn't cached.
+              return
+            }
+            // the body url is the request url plus 'include'
+            const url = new URL(request.url);
+            url.searchParams.set('i', 'i'); // Adds ?i=i or &i=i, which is our indicator for "internal" partial page
+            const startFetch = Promise.resolve(cachedShellTop);
+            const endFetch = Promise.resolve(cachedShellBottom);
```

Comment:
I didn't feel like re-fetching from the cache (`caches.match('/shell_top')`), so I just wrapped the values from earlier in a resolved promise. If you think that it's more clear/expressive to call `caches.match` again I'm happy to change